runtime_scan/pkg/provider/aws/client.go

```diff
@@ -797,6 +810,54 @@ func (c *Client) getInstancesFromDescribeInstancesOutput(ctx context.Context, re
 	return ret
 }
 
+type rootVolumeInfo struct {
+	DeviceName string
+	SizeGB     int32
+	Encrypted  bool
+}
+
+func getRootVolumeInfo(ctx context.Context, client *ec2.Client, i ec2types.Instance, region string) (*rootVolumeInfo, error) {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+	for _, mapping := range i.BlockDeviceMappings {
+		if getStringPointerValOrEmpty(mapping.DeviceName) == getStringPointerValOrEmpty(i.RootDeviceName) {
```

Comment:
To avoid weird cases like RootDeviceName is not set or empty, I would check
```
 if i.RootDeviceName == nil {
  return nil, fmt.Errorf("rootDevice not set")
}

```
at the beginning of the function

Reply:
good point but this is an internal function that runs right after validateInstanceFields. I think that we should validate it there. WDYT?
Do you think that better not to know about this asset in discovery (as validation failed) or only have no root volume info of (validate like you suggested)